src/Nethermind/Nethermind.Consensus/Processing/BlockCachePreWarmer.cs

```diff
@@ -207,20 +215,14 @@ private void WarmupAddresses(ParallelOptions parallelOptions, Block block)
         {
             if (parallelOptions.CancellationToken.IsCancellationRequested) return;
 
+            IReadOnlyTxProcessorSource env = PreWarmer._envPool.Get();
             try
             {
-                if (SystemTxAccessList is not null)
+                // Only need one scope as we are taking a parallel safe path
```

Comment:
Can you explain more, we are doing multi-threaded access to that instance?

Reply:
Yes

However now `Warm(Address)` and `Warm(AccessList, , isParallelAccess: true)` only use the ConcurrentDictionary and Trie in readonly mode; so are safe for multi-thread access skipping all data structures in the scope (as they would have been reset anyway)

i.e.
1. is in ConcurrentDict return
2. Get from trie and put in ConcurrentDict

When `Warm(AccessList, , isParallelAccess: false)` like in transaction; it also fills out the data structures in the scope as they are likely needed by the tx